python/tank/commands/dump_config.py

```diff
@@ -168,12 +168,28 @@ def _run(self, log, params):
         :param params: parameter dict.
         """
 
+        log.info("Dumping config...")
+
         # the env to dump
         env = self.tk.pipeline_configuration.get_environment(
             params["env"],
             writable=True
         )
 
+        # make sure the environment file doesn't match the output file.
+        # at some point we may want to make this possible, but dump_config is
+        # more about debugging, so for now just error out if they're the same file.
+        if params["file"]:
+            out_file = os.path.realpath(params["file"])
+            env_file = os.path.realpath(env.disk_location)
+
+            if out_file == env_file:
```

Comment:
I think sending both through `realpath` should normalize them. I think it should be ok. maybe... If you can think of a scenario let me know.